cmds/calc-default-gateway/main.go

```diff
@@ -0,0 +1,43 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/subspacecommunity/subspace/util"
+)
+
+var help *bool = flag.Bool("h", false, "show help and exit")
+var version string
+
+func usage() {
+	_, _ = fmt.Fprintf(os.Stderr, `Usage:
+  %[1]s [-h]
+  %[1]s CIDR
+Flags:
+`, filepath.Base(os.Args[0]))
+	flag.PrintDefaults()
+	_, _ = fmt.Fprintf(os.Stderr, "Version: %s", version)
+}
+
+func main() {
+	var err error
+	flag.Parse()
+	if *help {
+		usage()
+		return
+	}
+	if flag.NArg() != 1 {
+		usage()
+		os.Exit(-1)
+	}
+	cidr := flag.Arg(0)
+	gw, err := util.CalcDefaultGateway(cidr)
+	if err != nil {
+		_, _ = fmt.Fprintf(os.Stderr, "Failed to calc default gw: %v\n", err)
+		os.Exit(-1)
+	}
+	fmt.Printf("%s\n", gw.String())
```

Comment:
surely this can be implemented into the current main.go without having to be a completely seperate binary?

I state this as some are using the command `subspace` directly and this will not support them. I like the use of the `util` piece though, as i feel it can be used in the original `subspace` piece which will eventually have a golang standard project layout (its something i've adopted, as the format was layed out by others before me)

Reply:
Yes, we still can use `subspace` command directly.

It it because this separated binary is for calculating default gateway in `entrypoint.sh` only used in Docker.

Here:
https://github.com/subspacecommunity/subspace/blob/46eb3bbaca1aa1325277b8edab63b5aaaee4a2db/entrypoint.sh#L49-L57

I googled some command to calculate the default gateway for arbitrary CIDR, but I can't find it.

Reply:
OK, I will remove `util` package. Please wait a moment.